app/models/manageiq/providers/hawkular/middleware_manager/live_metrics_capture.rb

```diff
@@ -0,0 +1,135 @@
+module ManageIQ::Providers
+  class Hawkular::MiddlewareManager::LiveMetricsCapture
+    class TargetValidationError < RuntimeError; end
+    class MetricValidationError < RuntimeError; end
+
+    def initialize(target)
+      @target = target
+      unless @target.kind_of?(MiddlewareServer)
+        raise TargetValidationError, "Validation error: unknown target"
+      end
+      @ems = @target.ext_management_system
+    end
+
+    def metrics_available
+      resource = Struct.new(:id, :feed).new
+      resource.id = @target.nativeid
+      resource.feed = @target.feed
+      available = []
+      @ems.metrics_resource(resource).each do |metric|
+        available << {
+          :id   => metric.id,
+          :name => @target.class::METRICS_HWK_MIQ[metric.name],
+          :type => metric.type,
+          :unit => metric.unit
+        }
+      end
+      available
+    end
+
+    def validate_metric(metric)
+      if metric.nil?
+        false
+      elsif metric[:id].nil?
+        false
+      elsif metric[:name].nil?
+        false
+      elsif metric[:type].nil?
+        false
+      elsif metric[:unit].nil?
+        false
+      else
+        true
+      end
+    end
+
+    def metric(metric, start_time, end_time, interval)
```

Comment:
Having a method and a variable with the same name seems like i could be potentially confusing.  The variable seems like the right choice, so I'm not sure what the method name should be...what does this method do...`process_metric`? `collect_and_process_metric`?